op-chain-ops/cmd/geth/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+
+	"github.com/mattn/go-isatty"
+	"github.com/urfave/cli/v2"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/geth"
+	"github.com/ethereum-optimism/optimism/op-service/clock"
+	"github.com/ethereum-optimism/optimism/op-service/opio"
+	oprpc "github.com/ethereum-optimism/optimism/op-service/rpc"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/eth/ethconfig"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	gethrpc "github.com/ethereum/go-ethereum/rpc"
+)
+
+const version = "1.0.0"
+
+type ServerAPI struct {
+	c chan any
+}
+
+func (s *ServerAPI) Stop() {
+	s.c <- struct{}{}
+}
+
+func NewServerAPI(c chan any) *ServerAPI {
+	return &ServerAPI{
+		c: c,
+	}
+}
+
+func main() {
+	log.Root().SetHandler(log.StreamHandler(os.Stderr, log.TerminalFormat(isatty.IsTerminal(os.Stderr.Fd()))))
+
+	app := &cli.App{
+		Name:  "geth",
+		Usage: "Wrapper around a geth node",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  "deploy-config",
+				Usage: "Path to hardhat deploy config file",
+			},
+			&cli.StringFlag{
+				Name:  "outfile.l1",
+				Usage: "Path to L1 genesis output file",
+			},
+		},
+		Action: func(ctx *cli.Context) error {
+			deployConfig := ctx.String("deploy-config")
+			if len(deployConfig) == 0 {
+				return errors.New("Must specify a deploy-config")
+			}
+
+			config, err := genesis.NewDeployConfig(deployConfig)
+			if err != nil {
+				return err
+			}
+
+			// Temporary assertion on the clique signer address
+			if config.CliqueSignerAddress != (common.Address{}) {
+				return errors.New("Clique signer address must be empty")
+			}
+
+			if config.L1BlockTime == 0 {
+				log.Warn("Sanitizing L1 blocktime to 1")
+				config.L1BlockTime = 1
+			}
+
+			genesis, err := genesis.NewL1Genesis(config)
+			if err != nil {
+				return err
+			}
+
+			portSetter := func(ethCfg *ethconfig.Config, nodeCfg *node.Config) error {
+				nodeCfg.HTTPPort = 8545
+				return nil
+			}
+			l1Node, l1Backend, err := geth.InitL1Geth(config, genesis, clock.SystemClock, []*ecdsa.PrivateKey{}, portSetter)
+			if err != nil {
+				return err
+			}
+
+			if err := l1Node.Start(); err != nil {
+				return err
+			}
+			if err := l1Backend.Start(); err != nil {
+				return err
+			}
+
+			c := make(chan any)
+
+			rpcCfg := oprpc.ReadCLIConfig(ctx)
+			server := oprpc.NewServer(
+				rpcCfg.ListenAddr,
+				rpcCfg.ListenPort,
+				version,
+				oprpc.WithLogger(log.New("rpc_server")),
+			)
+			server.AddAPI(gethrpc.API{
```

Comment:
The idea here was to add a new rpc endpoint `admin_stop` so that we could remotely shut down this script. Open to better ideas but the flow is:
- run this command
- forge deployment script
- dump state
- call rpc to shut down this binary
- move state to location where op-e2e can pull it in